hackage-security/src/Hackage/Security/TUF/Snapshot.hs

```diff
@@ -68,6 +68,7 @@ instance MonadReader RepoLayout m => ToJSON m Snapshot where
 
 instance ( MonadReader RepoLayout m
          , MonadError DeserializationError m
+         , MonadFail m
```

Comment:
In this case I would probably try to use the `MonadError` constraint to instead report a more precise error. For instance, define locally:
```haskell
let lookupFile path =
      withExceptT DeserializationErrorSchema 
      $ liftEither $ FileMap.lookupM snapshotMeta path
```
and use this in place of `FIleMap.lookupM` below.

Reply:
I was not able to make this work so I just wrote:
```haskell
lookupErr :: MonadError e m => (String -> e) -> FileMap -> TargetPath -> m FileInfo
```